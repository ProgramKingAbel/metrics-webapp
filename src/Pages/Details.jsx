import React from 'react';
import { useSelector } from 'react-redux';

const Details = () => {
  const details = useSelector((state) => state.details.stock.historical);
  // eslint-disable-next-line
  console.log(details);

  return (
    <div>Details</div>
  );
};

export default Details;
