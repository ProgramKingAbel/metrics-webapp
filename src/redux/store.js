import { configureStore } from '@reduxjs/toolkit';
import stocksReducer from './features/stocks/stocksSlice';
import detailsReducer from './features/details/detailsSlice';

const store = configureStore({
  reducer: {
    stocks: stocksReducer,
    details: detailsReducer,
  },
});

export default store;
