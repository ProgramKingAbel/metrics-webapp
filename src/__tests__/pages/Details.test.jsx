import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/store';
import Details from '../../Pages/Details';

test('renders Details component correctly', () => {
  const { container } = render(<Provider store={store}><Router><Details /></Router></Provider>);
  expect(container).toMatchSnapshot();
});
