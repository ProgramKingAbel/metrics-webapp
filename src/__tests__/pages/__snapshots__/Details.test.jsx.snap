// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="btn btn-outline-primary"
      style="margin-top: 2rem;"
      type="button"
    >
      back
    </button>
    <div>
      Loading...
    </div>
    <div>
      Loading...
    </div>
  </div>
</div>
`;
